test/integration/user-management-api/services/proxies_test.rb

```diff
@@ -41,25 +43,64 @@ def test_crud_provider_key
   end
 
   def test_update
+    rolling_updates_on
+    rolling_update(:api_as_product, enabled: false)
+
     params = provider_key_params.merge(proxy: { endpoint: 'https://alaska.wild' })
 
     Proxy.update_all(apicast_configuration_driven: false)
-    Proxy.any_instance.expects(:deploy_v2).never
-    put(admin_api_service_proxy_path(params))
+    assert_difference @service.proxy.proxy_configs.sandbox.method(:count), 0 do
+      put(admin_api_service_proxy_path(params))
+    end
     assert_response :success
 
     Proxy.update_all(apicast_configuration_driven: true)
-    Proxy.any_instance.expects(:deploy_v2).once
-    put(admin_api_service_proxy_path(params))
+    assert_difference @service.proxy.proxy_configs.sandbox.method(:count), 1 do
+      put(admin_api_service_proxy_path(params))
+    end
+    assert_response :success
+  end
+
+  def test_update_apiap
+    rolling_updates_on
+    rolling_update(:api_as_product, enabled: true)
+
+    params = provider_key_params.merge(proxy: { endpoint: 'https://alaska.wild' })
+
+    Proxy.update_all(apicast_configuration_driven: false)
+    assert_difference @service.proxy.proxy_configs.sandbox.method(:count), 0 do
+      put(admin_api_service_proxy_path(params))
+    end
+    assert_response :success
+
+    Proxy.update_all(apicast_configuration_driven: true)
+    assert_difference @service.proxy.proxy_configs.sandbox.method(:count), 0 do
+      put(admin_api_service_proxy_path(params))
+    end
     assert_response :success
   end
 
   def test_update_for_service_mesh
     rolling_updates_on
+    rolling_update(:api_as_product, enabled: false)
+    
+    params = provider_key_params.merge(proxy: { credentials_location: 'headers' })
+    @service.update_column :deployment_option, 'service_mesh_istio'
+    
+    assert_difference @service.proxy.proxy_configs.sandbox.method(:count), 1 do
```

Comment:
For service mesh we expect the production to change too :)

Reply:
@hallelujah Then imo the ActiveDocs description is not entirely related with what it should do:
https://github.com/3scale/porta/blob/22b1411b8eda4b4d1e54818d89db209f39c9dbf6/app/controllers/admin/api/services/proxies_controller.rb#L36

Reply:
What should it say?

Reply:
For servcice mesh, it will also create a new APIcast configuration for production